lib/account/bloc/account_bloc.dart

```diff
@@ -43,9 +43,9 @@ class AccountBloc extends Bloc<AccountEvent, AccountState> {
   }
 
   Future<void> _refreshAccountInformation(RefreshAccountInformation event, Emitter<AccountState> emit) async {
-    add(GetAccountInformation());
-    add(GetAccountSubscriptions());
-    add(GetFavoritedCommunities());
+    await _getAccountInformation(GetAccountInformation(), emit);
+    await _getAccountSubscriptions(GetAccountSubscriptions(), emit);
+    await _getFavoritedCommunities(GetFavoritedCommunities(), emit);
```

Comment:
The problem here is that adding these things as events causes them to execute asynchronously, which creates a race condition (specifically because calculating the favorites requires having all of the subscriptions first). By specifically invoking and awaiting, we ensure that everything executes in the right order.